pkg/apis/core/types_shoot.go

```diff
@@ -154,6 +154,18 @@ type ShootStatus struct {
 	MigrationStartTime *metav1.Time
 	// Credentials contains information about the shoot credentials.
 	Credentials *ShootCredentials
+	// LastMaintenance holds information about the last maintenance operations on the Shoot.
+	LastMaintenance *LastMaintenance
+}
+
+// LastMaintenance holds information about a maintenance operation on the Shoot.
+type LastMaintenance struct {
+	// A human-readable message containing details about the operations performed in the last maintenance.
+	Description string
+	// TriggeredTime is the time when maintenance was triggered.
+	TriggeredTime metav1.Time
+	// Status holds details of last maintenance failure.
+	Status *string
```

Comment:
Let's rename this to `State` (like in `LastOperation`).

Also, I'm wondering whether we should use `LastOperationState` type here to have only `Error`/`Succeeed`/etc. For the error message in case maintenance fails, maybe we should add a `Reason` or `FailureReason` field?
Alternatively, we keep it as `string` (why a pointer, btw?) and only use "short messages" like `Succeeded` or `Failed: <error>`?

Reply:
I have kept only one field cause we can't populate if the maintenance succeded successfully in a consistent way. 
The `status` is a pointer cause it can be nil if the dry run of maintenance doesn't fail as there will be nothing to put in it.

Reply:
I have updated the PR accordingly PTAL d61810532444c44d112305fd85262f5e16b255f2.